Support stack versions greater than 0.1.7.0

wayofthepie opened the pull request:
Fixes rubik/stack-hpc-coveralls#4.

Updates the version check to support stack versions greater than **0.1.7.0**, also updates the travis config to test both **0.1.7.0** and **0.1.8.0** against **ghc-7.8.4** and **ghc-7.10.2**.

wayofthepie committed:
Support stack versions greater than 0.1.7.0

wayofthepie committed:
Allow travis to run on other branches temporarily

wayofthepie committed:
Fix version check, now uses --numeric-version

rubik commented on src/SHC/Utils.hs:
```diff
@@ -54,7 +54,7 @@ getRemotes = nubBy ((==) `on` name) <$> parseRemotes <$> git ["remote", "-v"]
 -- | Verify that the required Stack is present.
 checkStackVersion :: IO Bool
-checkStackVersion = ("Version 0.1.7" `isPrefixOf`) <$> stack ["--version"]
+checkStackVersion = ("0.1.7.0" <) <$> stack ["--numeric-version"]
```
Oh I didn't know about `--numeric-version`! But shouldn't that be a `(<=)`? Even if it's an odd version number and therefore a development version, SHC still works with it so we should allow it.

wayofthepie replied:
Wow, yes it should be! Will update.

rubik commented on .travis.yml:
```diff
@@ -17,11 +20,14 @@ cache:
+        curl -L https://github.com/rubik/argon/releases/download/test-stack-1.7/stack-linux-x64.tar.bz2 | tar -xj -C $HOME/.local/bin;
+    else
+        curl -L https://github.com/commercialhaskell/stack/releases/download/v$STACK_VERSION/$STACK_FULLVERSION.tar.gz | tar -xz -C /var/tmp;
+        mv /var/tmp/$STACK_FULLVERSION/stack $HOME/.local/bin;
```
Why do you prefer unpacking to `/tmp` and then moving to `bin/` when we can just unpack in the current directory? Currently release archives just contain the binary and nothing else.

wayofthepie replied:
The 1.8.0 release has docs as well:

``` bash
$ tar -tf stack-0.1.8.0-linux-x86_64.tar.gz 
stack-0.1.8.0-linux-x86_64/stack
stack-0.1.8.0-linux-x86_64/CONTRIBUTING.md
stack-0.1.8.0-linux-x86_64/ChangeLog.md
stack-0.1.8.0-linux-x86_64/GUIDE.md
stack-0.1.8.0-linux-x86_64/LICENSE
stack-0.1.8.0-linux-x86_64/README.md
stack-0.1.8.0-linux-x86_64/doc/GUIDE.md
stack-0.1.8.0-linux-x86_64/doc/MAINTAINER_GUIDE.md
stack-0.1.8.0-linux-x86_64/doc/SIGNING_KEY.md
stack-0.1.8.0-linux-x86_64/doc/architecture.md
stack-0.1.8.0-linux-x86_64/doc/build_command.md
stack-0.1.8.0-linux-x86_64/doc/dependency_visualization.md
stack-0.1.8.0-linux-x86_64/doc/docker_integration.md
stack-0.1.8.0-linux-x86_64/doc/faq.md
stack-0.1.8.0-linux-x86_64/doc/install_and_upgrade.md
stack-0.1.8.0-linux-x86_64/doc/nonstandard_project_init.md
stack-0.1.8.0-linux-x86_64/doc/shell_autocompletion.md
stack-0.1.8.0-linux-x86_64/doc/stack_and_ghcjs.md
stack-0.1.8.0-linux-x86_64/doc/yaml_configuration.md
```

It also has a path prefix when extracted by default, I can update to: 

``` bash
curl -L https://github.com/commercialhaskell/stack/releases/download/v$STACK_VERSION/$STACK_FULLVERSION.tar.gz |\
 tar xz --strip-components 1 -C ~/.local/bin $STACK_FULLVERSION/stack
```

Which will just extract and store **stack** directly in **~/local/bin**?

rubik replied:
Ah I see! Yes I think the second version is better, so that we just extract what's needed. It'd be great if you updated the PR, then I'll merge!

wayofthepie replied:
Ok done! Not sure if travis will like that "\", we'll see.

Edit : ya it doesn't like that "\", updated to run on a single line.

wayofthepie committed:
Fix version comparison

wayofthepie committed:
Only extract what's needed from stack release

wayofthepie committed:
Run curl on single line

Travis will try to run commands on a single line, so using "\" for
formatting will break the build.

rubik commented:
Perfect, thank you so much!
